rules/server/server.go

```diff
@@ -0,0 +1,129 @@
+// Package rulesserver provides primitives to interact with the openapi HTTP API.
+//
+// Code generated by github.com/deepmap/oapi-codegen version v1.8.3 DO NOT EDIT.
+package rulesserver
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/deepmap/oapi-codegen/pkg/runtime"
+	"github.com/go-chi/chi/v5"
+)
+
+// ServerInterface represents all server handlers.
+type ServerInterface interface {
+	// lists all rules for a tenant
+	// (GET /api/v1/{tenant}/rules)
+	ListRules(w http.ResponseWriter, r *http.Request, tenant string)
+	// set/overwrite the rules for the tenant
+	// (PUT /api/v1/{tenant}/rules)
+	SetRules(w http.ResponseWriter, r *http.Request, tenant string)
+}
+
+// ServerInterfaceWrapper converts contexts to parameters.
+type ServerInterfaceWrapper struct {
+	Handler            ServerInterface
+	HandlerMiddlewares []MiddlewareFunc
+}
+
+type MiddlewareFunc func(http.HandlerFunc) http.HandlerFunc
+
+// ListRules operation middleware
+func (siw *ServerInterfaceWrapper) ListRules(w http.ResponseWriter, r *http.Request) {
```

Comment:
There's something funny going on here. One of the nice things of deepmap-oapi-gen is that it can generate two server interfaces for you, a basic HTTP server and also a typed, domain-aware server that unmarshals all of the requests into the correct Golang type you defined in your OpenAPI spec. Any idea why this isn't getting generated?

In any case, if we _did_ generate this, then the generated server would _have_ to be in the same package as the types (and this also the client), otherwise the server would't know what types to use.

Maybe this nice server isn't getting generated because you've put the server into a separate package?

I think it would be very worth dealing with the nicer server interface. This makes the generated server much more useful, as it automatically checks and unpacks all of the request parameters for you and puts them into the correct types, etc. This is also how grpc clients/servers are normally pacakged, so it's not an unfamiliar pattern